hs_dataverse/utils.py

```diff
@@ -0,0 +1,672 @@
+          "typeClass": "primitive",
+          "value": "Consortium of Universities for the Advancement of Hydrological Science, Inc (CUAHSI)"
+        },
+        "distributorAbbreviation": {
+          "typeName": "distributorAbbreviation",
+          "multiple": false,
+          "typeClass": "primitive",
+          "value": "HydroShare"
+        },
+        "distributorURL": {
+          "typeName": "distributorURL",
+          "multiple": false,
+          "typeClass": "primitive",
+          "value": "http://www.hydroshare.org"
+        },
+        "distributorLogoURL": {
+          "typeName": "distributorLogoURL",
+          "multiple": false,
+          "typeClass": "primitive",
+          "value": "https://www.hydroshare.org/static/img/logo-lg.png"
+        }
+    }
+
+    software_dict = {
+        "softwareName": {
+          "typeName": "softwareName",
+          "multiple": false,
+          "typeClass": "primitive",
+          "value": ""
+        },
+        "softwareVersion": {
+          "typeName": "softwareVersion",
+          "multiple": false,
+          "typeClass": "primitive",
+          "value": ""
+        }
+    }
+
+    geo_coverage_dict = {
+            "country": {
+              "typeName": "country",
+              "multiple": false,
+              "typeClass": "controlledVocabulary",
+              "value": ""
+            },
+            "state": {
+              "typeName": "state",
+              "multiple": false,
+              "typeClass": "primitive",
+              "value": ""
+            },
+            "city": {
+              "typeName": "city",
+              "multiple": false,
+              "typeClass": "primitive",
+              "value": ""
+            },
+            "otherGeographicCoverage": {
+              "typeName": "otherGeographicCoverage",
+              "multiple": false,
+              "typeClass": "primitive",
+              "value": ""
+            }
+    }
+
+    bounding_box_dict = {
+                "westLongitude": {
+                  "typeName": "westLongitude",
+                  "multiple": false,
+                  "typeClass": "primitive",
+                  "value": ''
+                },
+                "eastLongitude": {
+                  "typeName": "eastLongitude",
+                  "multiple": false,
+                  "typeClass": "primitive",
+                  "value": ''
+                },
+                "northLongitude": {
+                  "typeName": "northLongitude",
+                  "multiple": false,
+                  "typeClass": "primitive",
+                  "value": ''
+                },
+                "southLongitude": {
+                  "typeName": "southLongitude",
+                  "multiple": false,
+                  "typeClass": "primitive",
+                  "value": ''
+                }
+    }
+
+    # Using the tags, extract fields into local variables
+    title = set_field(root.find(".//%s" % title_tag))
```

Comment:
all this stuff would go away with a templated solution. You wouldn't need to search for tags.